documentation/asciidoc/computers/configuration/raspi-config.adoc

```diff
@@ -247,364 +239,404 @@ Update this tool to the latest version.
 [[about]]
 ==== About raspi-config
 
-Selecting this option shows the following text:
-
-----
-This tool provides a straightforward way of doing initial configuration of the Raspberry Pi.
-Although it can be run at any time, some of the options may have difficulties if you have heavily customised your installation.
-----
+Selecting this option displays a description of `raspi-config`.
 
 [[finish]]
 ==== Finish
 
-Use this option when you have completed your changes. You will be asked whether you want to reboot or not. When implementing changes for the first time, it's best to reboot. There will be a delay in rebooting if you have chosen to resize your SD card.
+Use this option when you have completed your changes. You will be asked whether you want to reboot or not. When implementing changes for the first time, it's best to reboot. If you chose to resize your SD card, rebooting may take longer than usual.
 
 [[raspi-config-cli]]
-== The `raspi-config` Command Line Interface
+== The `raspi-config` CLI
 
-The `raspi-config` tool can also be run in a non-interactive mode, which is useful for setting up a Raspberry Pi image for distribution.
+The `raspi-config` tool supports non-interactive command line options and flags instead of the interactive menu. These options can be useful when configuring a Raspberry Pi image for distribution.
 
+[source,console]
 ----
-sudo raspi-config nonint <command> <arguments>
+$ sudo raspi-config nonint <command> <arguments> [optional-argument]
 ----
 
-The `sudo` is required because you will be changing files that are not owned by the default user.
-
-NOTE: There is no consistent meaning for `0` and `1` in arguments. Each function will document what `0` and `1` mean for that particular function.
+NOTE: The meaning of `0` and `1` varies between options. Always check the documentation before passing a value to an option.
 
 [[raspi-config-cli-commands]]
 === List of options
 
-NOTE: The `raspi-config` tool is under continuous development, and the list of options below may not be completely up to date. Please be aware that different models of Raspberry Pi may have different options available.
-
 ==== System options
 
 ===== Wireless LAN
 
 Allows setting of the wireless LAN SSID and passphrase.
 
+[source,console]
 ----
-sudo raspi-config nonint do_wifi_ssid_passphrase <ssid> <passphrase> [hidden] [plain]
+$ sudo raspi-config nonint do_wifi_ssid_passphrase <ssid> <passphrase> [hidden] [plain]
```

Comment:
Perhaps we should remove the `raspi-config nonint do_wifi_ssid_passphrase` documentation, and just refer people to the `nmcli` docs instead? https://www.raspberrypi.com/documentation/computers/configuration.html#using-the-command-line :shrug:

Reply:
I tested it today and it does work. Some of the errors are a bit wonky when you use nmcli, though -- I suspect some code paths might assume `wpa_supplicant` exists even when using nmcli.

I'm tempted to just point users to nmcli because of these issues, but as long as the option exists and works, some users might prefer it.

@spl237 , should we continue supporting this option? I'd be happy to open an issue on the `raspi-config` repo to either remove it or fix it. Your call.

Reply:
It's not something I ever use in the GUI; I think it is there for the benefit of people who use raspi-config from the command-line. So @XECDesign's call, not mine!

Reply:
What wonky error messages are you seeing and what's not working? If there's an issue I can reproduce, I'm happy to fix it.

However, I've always said that the `nonint` functionality is not intended for the user, so it doesn't make much sense documenting it. The idea is there is that it's used by other Raspberry Pi tools so that if we switch from dhcpcd to nm or make any other change, we only need to support it in one place. That way other tools won't need to be rebuilt and they won't diverge in behaviour. However, individual nonint commands can change or be removed without notice.

The short version - in my opinion, yes it absolutely makes sense for the user to use nmcli directly.